src/tir/transforms/lower_sparse_tir.cc

```diff
@@ -33,6 +33,38 @@
 namespace tvm {
 namespace tir {
 
+Map<Var, Buffer> UpdateBufferMap(PrimFunc f) {
+  struct BufferMapUpdater : public StmtVisitor {
+    explicit BufferMapUpdater(Map<Var, Buffer> buffer_map) : buffer_map_(std::move(buffer_map)) {}
+
+    void VisitStmt_(const SparseBlockNode* sp_block) {
+      for (const auto& it : sp_block->sp_struct2param_map) {
```

Comment:
Better add some type annotations, hard to read the `auto`'s here.

Reply:
Ah actually it's simply a Map iterator - just imagine you're iterating a `std::map`.

Reply:
Yes I understand, I just mean in this case the code is hard to read. Let's refactor later.